zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentResult.kt

```diff
@@ -24,4 +24,24 @@ class CreateDocumentResult(
     val bestandsnaam: String,
     val bestandsomvang: Long,
     val beginRegistratie: LocalDateTime,
-)
\ No newline at end of file
+    val bestandsdelen: List<Bestandsdelen>?
+) {
+    fun getBestandsdelenIdFromUrl(): String {
+        if (bestandsdelen.isNullOrEmpty()) {
+            return ""
+        }
+        return bestandsdelen[0].url.substring(bestandsdelen[0].url.lastIndexOf("/") + 1)
+    }
+
+    fun getLockFromBestanddelen(): String {
+        if (bestandsdelen.isNullOrEmpty()) {
+            return ""
+        }
+
+        return bestandsdelen[0].lock
```

Comment:
Does the spec say the lock is always the same for every bestandsdeel?

Reply:
It is a lock of the related informatieobject, not of the bestandsdeel. So yes it is the same for every bestandsdeel

lock required | string (lock)
Hash string, which represents id of the lock of related informatieobject

Reply:
I will debug the application for a definitive answer

Reply:
<img width="824" alt="image" src="https://github.com/user-attachments/assets/dd7b62a5-0bd6-4d97-bacb-6febba310b70">

Yes it is the same lock value for each bestandsdeel